internal/product/service.go

```diff
@@ -4,10 +4,14 @@ package product
 import (
 	"context"
 	"kg/procurement/internal/common/database"
+
+	"github.com/benbjohnson/clock"
 )
 
 type productDBAccessor interface {
 	GetProductsByVendor(ctx context.Context, vendorID string, spec GetProductsByVendorSpec) ([]Product, error)
+	UpdatePrice(ctx context.Context, price Price) (*Price, error)
+	UpdateProduct(ctx context.Context, payload Product) (*Product, error)
```

Comment:
Return object-nya langsung aja
```suggestion
	UpdatePrice(ctx context.Context, price Price) (Price, error)
	UpdateProduct(ctx context.Context, payload Product) (Product, error)
```